packages/protocol/src/action-protocol/client-notification.ts

```diff
@@ -13,53 +13,113 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-import { isNumber, isString } from '../utils/typeguard-util';
-import { Action, isActionKind } from './base-protocol';
+import { hasStringProp } from '../utils/type-util';
+import { Action } from './base-protocol';
 
-export type ServerSeverity = 'NONE' | 'INFO' | 'WARNING' | 'ERROR' | 'FATAL' | 'OK';
 /**
- * This action is typically sent by the server to signal a state change. This action extends the corresponding Sprotty action to include
- * a timeout. If a timeout is given the respective status should disappear after the timeout is reached.
+ * This action is typically sent by the server to signal a state change.
+ * If a timeout is given the respective status should disappear after the timeout is reached.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `ServerStatusActions`.
  */
-export class GLSPServerStatusAction implements Action {
-    static KIND = 'serverStatus';
-
-    constructor(
-        public severity: ServerSeverity,
-        public message: string,
-        public timeout = -1,
-        readonly kind = GLSPServerStatusAction.KIND
-    ) {}
+export interface ServerStatusAction extends Action {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.